[MRG] MAINT: remove old scipy LBFGS scipy related code

lesteve opened the pull request:
Very similar to https://github.com/scikit-learn/scikit-learn/pull/10704.

maxiter parameter and nit in returned info dict have been available in LBFGS starting in scipy 0.12.

lesteve committed:
MAINT: remove old scipy related code

maxiter parameter and nit in returned info dict have been available in LBFGS
starting in scipy 0.12.

rth commented:
The second part of the second sentence in
> 'lsqr' uses the dedicated regularized least-squares routine scipy.sparse.linalg.lsqr. It is the fastest but may not be available in old scipy versions. 

[here](https://github.com/scikit-learn/scikit-learn/blob/7700b5ac7343a57492e30fd310cc63e8fd653bfa/sklearn/linear_model/ridge.py#L246) can also be removed or updated. lsqr [was added in scipy 0.8](https://scipy.github.io/devdocs/release.0.8.0.html#new-sparse-least-squares-solver). 

LGTM.

lesteve committed:
Finish cleanup of scipy 0.13.3 move

lesteve committed:
lsqr has been available since scipy 0.8

lesteve committed:
Remove numpy < 1.5 related code

lesteve commented on sklearn/linear_model/ridge.py:
```diff
@@ -244,8 +244,8 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         - 'lsqr' uses the dedicated regularized least-squares routine
-          scipy.sparse.linalg.lsqr. It is the fastest but may not be available
-          in old scipy versions. It also uses an iterative procedure.
+          scipy.sparse.linalg.lsqr. It is the fastest and uses an iterative
```
Not sure what to write here, suggestions more than welcome! I guess "it is the fastest" without more details is not something I would trust and I am certainly no expert on this.

lesteve commented:
OK I removed more code related to old numpy/scipy versions that we don't support anymore.

lesteve committed:
Set random_state in test

random_state parameter is available in scipy.sparse.rand for the scipy versions we support.

lesteve committed:
random_state available in scipy.sparse.rand for all the scipy versions we support

jnothman reviewed (approved):
This is fine. I agree with your reservations about "it is the fastest" but whatever

jnothman commented:
Thanks
